compiler/rustc_hir_typeck/src/pat.rs

```diff
@@ -165,7 +165,13 @@ enum MutblCap {
 
 impl MutblCap {
     fn cap_mutbl_to_not(self, span: Option<Span>) -> Self {
-        if self == MutblCap::Mut { MutblCap::Not(span) } else { self }
+        if let Some(s) = span
+            && self != MutblCap::Not(None)
+        {
+            MutblCap::Not(Some(s))
+        } else {
+            MutblCap::Not(None)
+        }
```

Comment:
It seems like whether or not the `Span` is present is important in some way. Can you document exactly what this means?